src/vm/op_contract.rs

```diff
@@ -284,19 +296,36 @@ impl InstructionSet for ContractOp {
                 };
                 regs.set_n(RegA::A64, *reg, state.map(|s| s.value.as_u64()));
             }
-            ContractOp::LdG(state_type, index, reg) => {
+            ContractOp::LdG(state_type, reg_32, reg_s) => {
+                let Some(reg_32) = *regs.get_n(RegA::A8, *reg_32) else {
+                    fail!()
+                };
+                let index: u8 = reg_32.into();
+
                 let Some(state) = context
                     .global
                     .get(state_type)
-                    .and_then(|a| a.get(*index as usize))
+                    .and_then(|a| a.get(index as usize))
                 else {
                     fail!()
                 };
-                regs.set_s(*reg, Some(state.value.as_inner()));
+                regs.set_s(*reg_s, Some(state.value.as_inner()));
             }
-            ContractOp::LdC(_state_type, _index, _reg) => {
-                // TODO: implement global contract state
-                fail!()
+
+            ContractOp::LdC(state_type, reg_32, reg_s) => {
+                let Some(reg_32) = *regs.get_n(RegA::A16, *reg_32) else {
+                    fail!()
+                };
+                let index: u8 = reg_32.into();
+
+                let Some(state) = context
+                    .global
```

Comment:
Right now this command does exactly the same thing as the other (`LdG` and `LdC`). However the difference must be that the first takes a global state under a given index from the specific operation, while the other - from the global state.

I can't check the context object (`OpInfo`) atm, but if it lacks the differentiation of these two types of global state, we need to add it there

Reply:
Hi, sorry for delay.

Confirming, today we only get the global state of the operation.  I think we need to add the `ContractState` in OpInfo, right?

Reply:
It seems so. Let me know if you need a help with that.